abixen-platform-core/src/main/java/com/abixen/platform/core/controller/common/AbstractUserController.java

```diff
@@ -115,6 +124,30 @@ public FormValidationResultDto updateUser(@PathVariable Long id, @RequestBody @V
         return new FormValidationResultDto(userFormResult);
     }
 
+    @JsonView(WebModelJsonSerialize.class)
+    @RequestMapping(value = "/{id}/avatar", method = RequestMethod.GET)
+    public ResponseEntity<byte[]> getUserAvatar(@PathVariable Long id) throws IOException {
+        User user = userService.findUser(id);
+        InputStream in = new FileInputStream(platformResourceConfigurationProperties.getAvatarLibraryDirectory() + user.getHashKey());
+
+        byte[] b = IOUtils.toByteArray(in);
+
+        in.close();
+
+        final HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.IMAGE_PNG);
+
+        return new ResponseEntity<byte[]>(b, headers, HttpStatus.CREATED);
+
+    }
+
+    @JsonView(WebModelJsonSerialize.class)
+    @RequestMapping(value = "/{id}/avatar", method = RequestMethod.POST)
+    public String updateUserAvatar(@PathVariable Long id, @RequestParam("file") MultipartFile avatarFile) throws IOException {
+        userService.changeUserAvatar(id,avatarFile);
+        return null;
```

Comment:
Looks strange. Return User, then you don't have call once again for the updated user, because this object will have the new avatar hash name.